src/js/components/DataTable/stories/OnUpdate.js

```diff
@@ -0,0 +1,252 @@
+/* eslint-disable no-param-reassign */
+import React, { useCallback, useMemo, useState } from 'react';
```

Comment:
When I expand a group, the groups below it are no longer rendering.

https://user-images.githubusercontent.com/12522275/140793136-70a66324-4796-46e5-a348-d2f94c6eb2d7.mov

:

Reply:
It could be that the story just needs overflow="auto"?

Reply:
Its looking like InfiniteScroll has a defect where it calculates the index of the lastPage wrong if items.length is a multiple of step. It's getting 1 too large at the boundary causing it to not call onMore when needed. I'll submit a separate issue and PR for that after running it by Matt also